src/foam/blob/IdentifiedBlob.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foam.blob;
+
+import foam.core.X;
+
+import java.nio.Buffer;
+
+public class IdentifiedBlob
```

Comment:
We should generate this from the foam.CLASS so that serialization works.

Reply:
IdentifiedBlob has a property called delegate with a factory that looks up the delegate based on the ID. How would I handle that when auto generating?

Reply:
Add a javaFactory to the property to do the same thing